diff --git a/server/cypress/e2e/signup.cy.js b/server/cypress/e2e/signup.cy.js
index aa9e634..596775a 100644
--- a/server/cypress/e2e/signup.cy.js
+++ b/server/cypress/e2e/signup.cy.js
@@ -1,5 +1,5 @@
 import { faker } from '@faker-js/faker';
-import api from '../../src/models/static';
+import api from '../../app/frontend/models/static';
 
 describe('signup', () => {
   it('signup screen works', function () {
