commit 5ad56ac2701622c4d2f00ec7fd3c64db34a0357c
Author: JP Rosevear <jp@codalio.com>
Date:   Fri Sep 6 10:31:28 2024 -0400

    Docker

diff --git a/Dockerfile b/Dockerfile
index bdd99ab..f63e26c 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -24,12 +24,23 @@ FROM base AS build
 # Install packages needed to build gems
 RUN apt-get install --no-install-recommends -y build-essential git libpq-dev pkg-config
 
+# Install JavaScript dependencies
+ARG NODE_VERSION=20.14.0
+ENV PATH=/usr/local/node/bin:$PATH
+RUN curl -sL https://github.com/nodenv/node-build/archive/master.tar.gz | tar xz -C /tmp/ && \
+    /tmp/node-build-master/bin/node-build "${NODE_VERSION}" /usr/local/node && \
+    rm -rf /tmp/node-build-master
+
 # Install application gems
 COPY Gemfile Gemfile.lock ./
 RUN bundle install && \
     rm -rf ~/.bundle/ "${BUNDLE_PATH}"/ruby/*/cache "${BUNDLE_PATH}"/ruby/*/bundler/gems/*/.git && \
     bundle exec bootsnap precompile --gemfile
 
+# Install node modules
+COPY package.json package-lock.json ./
+RUN npm install --frozen-lockfile
+
 # Copy application code
 COPY . .
 
