diff --git a/.prettierignore b/.prettierignore
index 56dae809..8962463f 100644
--- a/.prettierignore
+++ b/.prettierignore
@@ -1 +1 @@
-src/models/static.js 
+app/frontend/src/models/static.js
diff --git a/app/frontend/tsconfig.json b/app/frontend/tsconfig.json
index 860b1d5f..18b5bc67 100644
--- a/app/frontend/tsconfig.json
+++ b/app/frontend/tsconfig.json
@@ -19,9 +19,9 @@
     "isolatedModules": true,
     "noEmit": true,
     "jsx": "react-jsx",
-    "baseUrl": "src"
+    "baseUrl": "./src"
   },
   "include": [
-    "src"
+    "./"
   ]
 }
diff --git a/app/views/layouts/application.html.erb b/app/views/layouts/application.html.erb
index 8aefb947..bca8a5d2 100644
--- a/app/views/layouts/application.html.erb
+++ b/app/views/layouts/application.html.erb
@@ -4,7 +4,8 @@
     <title><%= Rails.application.class.module_parent_name %></title>
     <%= csrf_meta_tags %>
     <%= vite_client_tag %>
-    <%= vite_javascript_tag 'application' %>
+    <%= vite_react_refresh_tag %>
+    <%= vite_javascript_tag 'index' %>
     <!--
       If using a TypeScript entrypoint file:
         vite_typescript_tag 'application'
diff --git a/vite.config.ts b/vite.config.ts
index 0f10a733..f709af41 100644
--- a/vite.config.ts
+++ b/vite.config.ts
@@ -1,5 +1,5 @@
 import { defineConfig, loadEnv, transformWithEsbuild } from 'vite';
-import { configDefaults } from 'vitest/config';
+import RubyPlugin from 'vite-plugin-ruby';
 import react from '@vitejs/plugin-react';
 import { ViteEjsPlugin } from 'vite-plugin-ejs';
 import eslint from 'vite-plugin-eslint';
@@ -22,7 +22,7 @@ function escapeRegExp(string) {
 
 const nodeModulesStr = `${path.sep}node_modules${path.sep}`;
 // NOTE: Keep trailing slash to use resulting path in prefix matching.
-const srcDir = url.fileURLToPath(new URL('./src/', import.meta.url));
+const srcDir = url.fileURLToPath(new URL('./app/frontend/', import.meta.url));
 // NOTE: Since ESBuild evaluates this regex using Go's engine, it is not
 // clear whether the JS-specific regex escape logic is sound.
 const srcRegex = new RegExp(`^${escapeRegExp(srcDir)}.*\.js$`);
@@ -99,6 +99,7 @@ export default defineConfig(({ mode }) => {
     },
 
     plugins: [
+      RubyPlugin(),
       vitePlugin(mode === 'production'),
       react(),
       {
