diff --git a/vite.config.ts b/vite.config.ts
index 58fab5d..1ac2ef3 100644
--- a/vite.config.ts
+++ b/vite.config.ts
@@ -43,7 +43,7 @@ export default defineConfig(({ mode }) => {
     test: {
       environment: 'jsdom',
       globals: true,
-      setupFiles: ['src/__tests__/shared/setupTests'],
+      setupFiles: ['__tests__/shared/setupTests'],
       // For now until speed improves https://dev.to/thejaredwilcurt/improving-vitest-performance-42c6
       testTimeout: 10000
     }
