diff --git a/server/vite.config.ts b/server/vite.config.ts
index b177068..58fab5d 100644
--- a/server/vite.config.ts
+++ b/server/vite.config.ts
@@ -3,6 +3,7 @@ import { defineConfig, loadEnv } from 'vite';
 import { ViteEjsPlugin } from 'vite-plugin-ejs';
 import eslint from 'vite-plugin-eslint';
 import { RhinoProjectVite } from '@rhino-project/vite-plugin-rhino';
+import RubyPlugin from 'vite-plugin-ruby'
 
 // https://vitejs.dev/config/
 export default defineConfig(({ mode }) => {
@@ -18,6 +19,7 @@ export default defineConfig(({ mode }) => {
     },
 
     plugins: [
+      RubyPlugin(),
       RhinoProjectVite({ enableJsxInJs: false }),
       react(),
       {
